account-registration/src/main/java/com/amazon/aws/spinnaker/plugin/registration/AccountsStatus.java

```diff
@@ -196,6 +209,7 @@ private Response getResourceFromApiGateway(String url) {
         if (this.headerGenerator == null) {
             makeHeaderGenerator(url);
             if (this.headerGenerator == null) {
+                log.error("Failed to generate resources required for AWS Signature V4 to authenticate with API Gateway.");
```

Comment:
This can possibly be simplified. logging the headerGenertor value in the function and also handling the exception so that it doesnt get repeated a few lines down.

Reply:
This is definitely something I'd like to get to for RFC compatible version.